### PR TITLE
feat: add plugin guides

### PR DESCRIPTION
**Pull request by UndiedGamer**
This Pull Request contains the guides for Plugins

### CODE REVIEW

**Comment by UndiedGamer** [comment]
need to convert this to draft

**Review by favna — CHANGES_REQUESTED** [review]

File: docs/Guide/plugins/plugin-api.mdx [hunk 1]
```diff
@@ -2,4 +2,83 @@
...
+</Tabs>
+
+Then, you can use the following configuration options in your SapphireClient extension class or initializer. This will
+either be located in your `new SapphireClient` constructor call, or `super` in your constructor method if you use an
+extension class.
+
+```javascript
+{
+  auth: {
+	// The application/client ID of your bot.
```
Comment (Lioness100) [inline docs/Guide/plugins/plugin-api.mdx:58]: This is a general comment but periods at the end of comments or lack thereof is not consistent throughout the guides & the documentation and it makes me a bit mad sometimes :')

File: docs/Guide/plugins/plugin-i18next.mdx [hunk 2]
```diff
@@ -2,4 +2,337 @@
...
+```
+
+The `resolveKey` function can be used anywhere to get translated text by its key. In this example, it is used in a
+method to send a message.
+
+<Tabs groupId="language-choice">
+<TabItem value="javascript" label="JavaScript" default>
+
+```javascript
+const { resolveKey } = require('@sapphire/plugin-i18next');
```
Comment (Lioness100) [inline docs/Guide/plugins/plugin-i18next.mdx:71]: Another consistency thing that drives me a bit mad throughout the guides & docs is tab indents vs two space indents :')

**Review by Lioness100 — CHANGES_REQUESTED** [review]

File: docs/Guide/plugins/plugin-i18next.mdx (hunk 2 above)
Comment (favna) [reply docs/Guide/plugins/plugin-i18next.mdx:71]: For the record: everything should be 2 space indents because otherwise it renders like crap in browsers.

Comment (UndiedGamer) [reply docs/Guide/plugins/plugin-i18next.mdx:71]: And I always format the code so I see no issue with spaces and tabs

File: docs/Guide/plugins/plugin-api.mdx (hunk 1 above)
Comment (UndiedGamer) [reply docs/Guide/plugins/plugin-api.mdx]: i cannot understand but can you explain?

**Comment by UndiedGamer** [comment]
the deploy isnt working, idk why but thanks to @Megatank58 we have a temporary deploy
https://3000-blush-louse-mfo46juj.ws-us18.gitpod.io/docs/Guide/

File: docs/Guide/plugins/plugin-api.mdx (hunk 1 above)
Comment (Lioness100) [reply docs/Guide/plugins/plugin-api.mdx]: Throughout the docs and guides, sometimes comments end with periods, and sometimes they don't. Just a little consistency thing.

**Review by kyranet — APPROVED** [review]
Was told the TODO pages are intentional, so the contents look good to me. 👍🏼

File: .vscode/settings.json [hunk 3]
```diff
@@ -15,5 +15,8 @@
 		"**/bower_components": true,
 		"**/node_modules": true,
 		"**/projects/": true,
-	}
+	},
+	"cSpell.words": [
+		"favna"
```
Comment (favna) [inline .vscode/settings.json:20]: kek

**Review by favna — APPROVED** [review]

**Pull request merged**